pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -947,8 +957,11 @@ self: super: {
     sha256 = "1yh2g45mkfpwxq0vyzcbc4nbxh6wmb2xpp0k7r5byd8jicgvli29";
   });
 
-  # Glob depends conditionally on semigroups for GHC < 8
-  Glob = if pkgs.lib.versionAtLeast self.ghc.version "8.0"
-    then super.Glob
-    else addBuildDepend super.Glob self.semigroups;
+
+  # packages that depend on extra packages for GHC <8.0
+  Glob = ifLT8 (p: addBuildDepend p self.semigroups) super.Glob;
+  # cereal must have `fail` in pre-ghc-8.0.x versions
+  # also tests require bytestring>=0.10.8.1
+  cereal = ifLT8 (p: dontCheck (addBuildDepend p self.fail)) super.cereal;
+  cereal_0_5_2_0 = ifLT8 (p: dontCheck (addBuildDepend p self.fail)) super.cereal_0_5_2_0;
```

Comment:
> Besides, it's not uncommon for those overrides to require even more extra build inputs the older the compiler gets

In that case the function can be applied multiple times with different compiler versions.
So should I copy and paste that into every 7.x configuration file?